pika/adapters/blocking_connection.py

```diff
@@ -344,6 +342,8 @@ def _handle_read(self):
         keeps track of how many frames have been written since the last read.
 
         """
+        if not hasattr(self, '_read_poller'):
+            raise exceptions.ConnectionClosed()
```

Comment:
Hi @wjps, just to clarify. I think that my two PRs (#533 and #542) make this "AttributeError" issue go away altogether. I could no longer reproduce it with those PRs in place.